rpc/handlers.go

```diff
@@ -996,3 +1000,26 @@ func (h *Handler) AddDeclareTransaction(declareTx json.RawMessage) (*DeclareTxRe
 func (h *Handler) Version() (string, *jsonrpc.Error) {
 	return h.version, nil
 }
+
+func (h *Handler) TransactionStatus(hash felt.Felt) (Status, *jsonrpc.Error) {
+	var status Status
+
+	receipt, txErr := h.TransactionReceiptByHash(hash)
+	if txErr == nil {
+		status = receipt.Status
+	} else if txErr == ErrTxnHashNotFound {
+		txStatus, err := h.feederClient.Transaction(context.Background(), &hash)
+		if err != nil {
+			return 0, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+		}
+
+		err = status.UnmarshalJSON([]byte(txStatus.Status))
+		if err != nil {
+			return 0, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+		}
```

Comment:
Let's not do this IMO. If we can't find the transaction in our db, we dont need to query the gateway.